modules/extensions-support/src/main/java/org/mule/runtime/module/extension/internal/loader/enricher/DynamicMetadataDeclarationEnricher.java

```diff
@@ -76,11 +78,11 @@ public void enrich(ExtensionLoadingContext extensionLoadingContext) {
 
     @Override
     public void enrich(ExtensionLoadingContext extensionLoadingContext) {
-      Optional<ExtensionTypeDescriptorModelProperty> property = extensionLoadingContext.getExtensionDeclarer().getDeclaration()
-          .getModelProperty(ExtensionTypeDescriptorModelProperty.class);
-
+      BaseDeclaration declaration = extensionLoadingContext.getExtensionDeclarer().getDeclaration();
       //TODO MULE-14397 - Improve Dynamic Metadata Enricher to enrich without requiring Classes
-      if (property.isPresent() && property.get().getType().getDeclaringClass().isPresent()) {
+      if (!isASTMode(declaration)) {
+        Optional<ExtensionTypeDescriptorModelProperty> property =
+            declaration.getModelProperty(ExtensionTypeDescriptorModelProperty.class);
         extensionType = property.get().getType();
```

Comment:
![Bug](https://sonarsource.github.io/sonar-github/bug.png 'Bug') Bug: Call "property.isPresent()" before accessing the value. ([squid:S3655](https://sonarqube.kbuild.msap.io/coding_rules#rule_key=squid%3AS3655))

[See it in SonarQube](https://sonarqube.kbuild.msap.io/project/issues?id=mule-runtime-mule&pullRequest=7618&issues=AWiF0bSNotIXnmsUlGnW&open=AWiF0bSNotIXnmsUlGnW)

[ISSUE_KEY]: AWiF0bSNotIXnmsUlGnW